hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/OverReplicatedProcessor.java

```diff
@@ -73,7 +73,7 @@ public void processAll() {
       try {
         processContainer(overRep);
         processed++;
-      } catch (IOException e) {
+      } catch (Exception e) {
```

Comment:
Should we handle Throwable to be safe? Unless service is not running, this thread must be running.

Reply:
I don't think `OutOfMemoryError` and other errors should be caught, so I think `Exception` is fine.

Reply:
I just looked at RM threads in HDFS and we catch throwable. I did not remember how that evolved. 
I think at least one instance I remember RM dead and NN was still running. ( Of course in a bad way).

We need to make sure process going down after this event.

```
@Override
    public void run() {
      while (namesystem.isRunning()) {
        try {
          // Process recovery work only when active NN is out of safe mode.
          if (isPopulatingReplQueues()) {
            computeDatanodeWork();
            processPendingReconstructions();
            rescanPostponedMisreplicatedBlocks();
          }
          TimeUnit.MILLISECONDS.sleep(redundancyRecheckIntervalMs);
        } catch (Throwable t) {
          if (!namesystem.isRunning()) {
            LOG.info("Stopping RedundancyMonitor.");
            if (!(t instanceof InterruptedException)) {
              LOG.info("RedundancyMonitor received an exception"
                  + " while shutting down.", t);
            }
            break;
          } else if (!checkNSRunning && t instanceof InterruptedException) {
            LOG.info("Stopping RedundancyMonitor for testing.");
            break;
          }
          LOG.error("RedundancyMonitor thread received Runtime exception. ",
              t);
          terminate(1, t);
        }
      }
    }
```

I am not saying to copy this, but this sustained and evolved from many such issues with similar use case in the system.

Reply:
Fair enough I will change this to throwable.